.github/workflows/sync-upstream-push.yml

```diff
@@ -0,0 +1,53 @@
+name: Sync with Upstream (push)
+
+on:
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    if: github.event.review.state == 'approved' && github.event.pull_request.user.login == 'gha-runner-images-updater-test[bot]'
+    steps:
+      - name: Generate GitHub App token
+        id: generate-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.GH_APP_ID }}
+          private-key: ${{ secrets.GH_APP_KEY }}
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.generate-token.outputs.token }}
+
+      - name: Configure Git
+        run: |
+          git config user.name 'gha-runner-images-updater-test[bot]'
+          git config user.email '172421971+gha-runner-images-updater-test[bot]@users.noreply.github.com@github.com'
+
+      - name: Check if changes will be removed from main
+        id: check-changes
+        run: |
+          git fetch origin
+          changest_removed=$(git range-diff origin/main...origin/${{ github.event.pull_request.head.ref }} | grep -q '<')
+          if [ -n "$changes_removed" ]; then
+            echo "Commits will be removed from main, this shouldn't happen. Please request changes on the pull request."
+            exit 1
+          else
+            echo "Commits will not be removed from main."
+            exit 0
+          fi
+
+      - name: Push changes to main
+        run: |
+          git push -f origin main
+
+      - name: Close pull request
```

Comment:
Why push and close? Can't we just merge?

Reply:
Because of the force push you can't use the buttons on the PR 🤔

Reply:
ooh yeah I was mistaken I think I've seen this happen with `gh`. maybe `gh` is not a bad thing to consider using 🤔

Reply:
https://cli.github.com/manual/gh_pr_merge

Reply:
It shouldn't be possible to merge if we go with the rebase approach. You would be merging commits that are already there.

Reply:
I also need to run some tests, but I just realized the close step might not even be necessary as the PR will be considered merged once all the commits are present in the target branch.

Reply:
Yup, the close step is not even necessary, the PR is automagically marked as merged.